apps/src/templates/sessions/LockoutPanel.jsx

```diff
@@ -0,0 +1,280 @@
+  // email in the field.
+  const [disabled, setDisabled] = useState(() => !isEmail(props.pendingEmail));
+
+  // When the email field is updated, also update the disability state of the
+  // submit button.
+  const onEmailUpdate = event => {
+    setDisabled(!isEmail(event.target.value));
+  };
+
+  // This will set the email to the current pending email and fire off the
+  // form as though they had typed in the same email again.
+  const resendPermissionEmail = event => {
+    // Don't submit the form... we will do that ourselves.
+    event.preventDefault();
+
+    const field = document.getElementById('parent-email');
+    field.value = props.pendingEmail;
+    this.submit();
+  };
+
+  // Get the current locale.
+  const locale = cookies.get('language_') || 'en-US';
+
+  // Whether or not we are rendering right-to-left.
+  const isRTL = getStore().getState()?.isRtl;
+
+  // How to format any localized dates on the page.
+  const dateOptions = {
+    year: 'numeric',
+    month: 'long',
+    day: 'numeric',
+  };
+
+  // We want to mark the email in the prompt in bold, so we split the
+  // localized string. It is a slight hack since we cannot just embed React
+  // component code into a string. Should work as though we could though.
+  const pendingPrompt = i18n.sessionLockoutPendingPrompt({
+    pendingEmail: '{pendingEmail}',
+  });
+  const pendingPromptParts = pendingPrompt.split('{pendingEmail}');
+
+  return (
+    <div style={styles.container} className="lockout-panel">
+      {/* Header image: Depends of if permission request is sent. */}
+      <img
+        style={styles.image}
+        src={props.pendingEmail ? headerThanksImage : headerImage}
+      />
+
+      <h2>
+        {props.pendingEmail
+          ? i18n.sessionLockoutPendingHeader()
+          : i18n.sessionLockoutNewAccountHeader()}
+      </h2>
+
+      {/* This form will post a permission request for the student.*/}
+      <form action={props.apiURL} method="post">
+        {/* The top prompt, which depends on whether or not a request is pending. */}
+        {props.pendingEmail && (
+          <p>
+            {pendingPromptParts[0]}
+            <strong>{props.pendingEmail}</strong>
+            {pendingPromptParts[1]}
+          </p>
+        )}
+        {!props.pendingEmail && <p>{i18n.sessionLockoutPrompt()}</p>}
+
+        {/* The timezone is set to UTC to ensure that the exact date renders. */}
+        <p>
+          {i18n.sessionLockoutNote({
+            deleteDate: props.deleteDate.toLocaleDateString(locale, {
+              ...dateOptions,
+              timeZone: 'UTC',
+            }),
+          })}
+        </p>
+
+        {/* This field shows the current status of the validation. */}
+        {/* Parent Permission: Not Pending / Pending */}
+        <div style={styles.statusSection}>
+          <label
+            style={isRTL ? styles.statusLabelRTL : styles.statusLabel}
+            htmlFor="permission-status"
```

Comment:
Oh cool, I didn't know you had to use `htmlFor` instead of `for` in JSX. That makes sense. Thanks for looking this up :)